python-mini-projects/projects/Web_page_summation/utils/utils.py

```diff
@@ -0,0 +1,118 @@
+import re
+import collections
+import pickle
+import numpy as np
+from nltk.tokenize import word_tokenize
+default_path = '.'
+train_article_path = '.'
+train_title_path = '.'
+valid_article_path = '.'
+
+
+def clean_str(sentence):
+    sentence = re.sub("[#.]+", "#", sentence)
+    return sentence
+
+
+def get_text_list(data_path, toy):
+    with open(data_path, "r", encoding="utf-8") as f:
+        if not toy:
+            return [clean_str(x.strip()) for x in f.readlines()][:200000]
+        else:
+            return [clean_str(x.strip()) for x in f.readlines()][:50]
+
+
+def build_dict(step, toy=False):
+    if step == "train":
+        train_article_list = get_text_list(train_article_path, toy)
+        train_title_list = get_text_list(train_title_path, toy)
+
+        words = list()
+        for sentence in train_article_list + train_title_list:
+            for word in word_tokenize(sentence):
+                words.append(word)
+
+        word_counter = collections.Counter(words).most_common()
+        word_dict = dict()
+        word_dict["<padding>"] = 0
+        word_dict["<unk>"] = 1
+        word_dict["<s>"] = 2
+        word_dict["</s>"] = 3
+        for word, _ in word_counter:
+            word_dict[word] = len(word_dict)
+
+        with open(default_path + "word_dict.pickle", "wb") as f:
+            pickle.dump(word_dict, f)
+
+    elif step == "valid":
+        with open(default_path + "word_dict.pickle", "rb") as f:
+            word_dict = pickle.load(f)
+
+    reversed_dict = dict(zip(word_dict.values(), word_dict.keys()))
+
+    article_max_len = 50
+    summary_max_len = 15
+
+    return word_dict, reversed_dict, article_max_len, summary_max_len
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are unpacking more than three variables while using a function that returns more than one variable. Using a large number of return values is prone to errors. We recommend that you return a small class or `namedtuple` instance instead.

[Learn more](https://docs.python.org/3/library/typing.html#typing.NamedTuple)